### PR TITLE
Enable useUnknownInCatchVariables

### PR DESCRIPTION
**Pull request by bhollis**
This enables (or rather, removes the suppression of) the `useUnknownInCatchVariables` TS option, and then fixes the errors that come from it. This helps our `catch` blocks no longer be `any` typed.

### CODE REVIEW

File: src/StorageTest.tsx [hunk 1]
```diff
@@ -32,16 +31,8 @@ export async function storageTest() {
   try {
     await set('idb-test', true);
   } catch (e) {
-    errorLog('storage', 'Failed IndexedDB Test - trying to delete database', e);
-    try {
-      await deleteDatabase();
-      await set('idb-test', true);
-      // Report to sentry, I want to know if this ever works
-      reportException('deleting database fixed IDB', e);
```
Comment (bhollis) [inline src/StorageTest.tsx:40]: I couldn't find any evidence in Sentry that this had ever been reported, so I removed the deletion behavior.

File: src/app/bungie-api/authenticated-fetch.ts [hunk 2]
```diff
@@ -32,8 +32,7 @@ export async function fetchWithBungieOAuth(
     const token = await getActiveToken();
     request.headers.set('Authorization', `Bearer ${token.accessToken.value}`);
   } catch (e) {
-    // Note: instanceof doesn't work due to a babel bug:
```
Comment (bhollis) [inline src/app/bungie-api/authenticated-fetch.ts:35]: This bug has long been fixed / it's possible to extend Error.

File: src/app/bungie-api/oauth.ts [hunk 3]
```diff
@@ -1,59 +1,59 @@
...
+  async (refreshToken: Token): Promise<Tokens> => {
     const body = new URLSearchParams({
       grant_type: 'refresh_token',
       refresh_token: refreshToken.value,
       client_id: oauthClientId(),
       client_secret: oauthClientSecret(),
     });
-    // https://github.com/zloirock/core-js/issues/178#issuecomment-192081350
-    // ↑ we return fetch wrapped in an extra Promise.resolve so it has proper followup methods
-    return Promise.resolve(
```
Comment (bhollis) [inline src/app/bungie-api/oauth.ts:20]: Another old browser bug. I took this opportunity to rewrite to async/await

File: src/app/utils/exceptions.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ if ($featureFlags.sentry) {
     sampleRate: $DIM_VERSION === 'beta' ? 0.5 : 0.01, // Sample Beta at 50%, Prod at 1%
     attachStacktrace: true,
     integrations: [
-      new TracingIntegrations.BrowserTracing({
+      new BrowserTracing({
```
Comment (bhollis) [inline src/app/utils/exceptions.ts:54]: The `@sentry/tracing` package has been deprecated.

File: src/app/bungie-api/authenticated-fetch.ts [hunk 5]
```diff
@@ -99,7 +98,11 @@ export async function getActiveToken(): Promise<Tokens> {
   try {
     return await getAccessTokenFromRefreshToken(token.refreshToken!);
   } catch (e) {
-    return await handleRefreshTokenError(e);
+    if (e instanceof Error || e instanceof Response) {
+      return await handleRefreshTokenError(e);
+    } else {
+      throw e;
+    }
```
Comment (robojumper) [inline src/app/bungie-api/authenticated-fetch.ts:105]: In the case of `e instanceof Error`, `handleRefreshTokenError` rethrows too, which is slightly confusing control flow -- maybe `handleRefreshTokenError` could also just take `unknown` and handle all rethrowing?

File: src/app/inventory/advanced-write-actions.ts [hunk 6]
```diff
@@ -299,7 +299,11 @@ async function getAwaToken(
       }
       */
     } catch (e) {
-      throw new DimError('AWA.FailedToken').withError(e);
+      const dimError = new DimError('AWA.FailedToken');
+      if (e instanceof Error) {
+        dimError.withError(e);
+      }
+      throw dimError;
```
Comment (robojumper) [inline src/app/inventory/advanced-write-actions.ts:306]: Why not `.withError(convertToError(e))` here?

File: src/app/loadout-drawer/loadout-apply.ts [hunk 7]
```diff
@@ -1400,14 +1405,18 @@ function applyMod(
...
+          error: errorMessage(e),
           item: item.name,
           plug: plugName,
         })
-      ).withError(e);
+      );
+      if (e instanceof Error) {
+        dimError.withError(e);
+      }
+      throw dimError;
```
Comment (robojumper) [inline src/app/loadout-drawer/loadout-apply.ts:1419]: Similarly, why not `.withError(convertToError(e))` here?

**Review by robojumper — COMMENTED** [review]
Getting new
```
Tracing extension 'startTransaction' has not been added. Call 'addTracingExtensions' before calling 'init':
Sentry.addTracingExtensions();
Sentry.init({...});
```
warnings in the console with this.

File: src/app/inventory/advanced-write-actions.ts (hunk 6 above)
Comment (bhollis) [reply src/app/inventory/advanced-write-actions.ts:306]: Because I'd already switched those all before I came up with `convertToError` :-)

**Review by robojumper — APPROVED** [review]

**Pull request merged**